bundle/edu.gemini.itc.shared/src/main/java/edu/gemini/itc/shared/ConfigCreator.java

```diff
@@ -0,0 +1,201 @@
+            step.putItem(SlitWidthKey, gnirsParams.slitWidth().logValue());
+
+            if (gnirsParams.altair().isDefined()) {
+                AltairParameters altairParameters = gnirsParams.altair().get();
+                step.putItem(AOGuideStarTypeKey, altairParameters.wfsMode().displayValue());
+                step.putItem(AOSystemKey, "Altair");
+                step.putItem(AOFieldLensKey, altairParameters.fieldLens().displayValue());
+            }
+        }
+        return conf;
+    }
+
+    public Config[] createNiriConfig(NiriParameters niriParams, int numExp) {
+        Config[] conf = createCommonConfig(numExp);
+
+        for (Config step : conf) {
+            step.putItem(ReadModeKey, (niriParams.readMode()));
+            step.putItem(InstInstrumentKey, "NIRI");
+            step.putItem(BuiltinROIKey, (niriParams.builtinROI()));
+            step.putItem(DisperserKey, (niriParams.grism().displayValue()));
+
+            if (niriParams.altair().isDefined()) {
+                AltairParameters altairParameters = niriParams.altair().get();
+                step.putItem(AOGuideStarTypeKey, altairParameters.wfsMode().displayValue());
+                step.putItem(AOSystemKey, "Altair");
+                step.putItem(AOFieldLensKey, altairParameters.fieldLens().displayValue());
+            }
+        }
+        return conf;
+    }
+
+    public Config[] createGmosConfig(GmosParameters gmosParams, int numExp) {
+        Config[] conf = createCommonConfig(numExp);
+
+        for (Config step : conf) {
+            System.out.println("SIte: " + gmosParams.site().displayName);
+            if (gmosParams.site().displayName.equals("Gemini North")) {
+                step.putItem(InstInstrumentKey, "GMOS-N");
+            } else if (gmosParams.site().displayName.equals("Gemini South")) {
+                step.putItem(InstInstrumentKey, "GMOS-S");
+            } else {
+                throw new Error("invalid site");
+            }
+
+            step.putItem(FPUKey, (gmosParams.fpMask().displayValue()));
+            step.putItem(AmpReadModeKey, (gmosParams.ampReadMode()));
+            step.putItem(DetectorManufacturerKey, (gmosParams.ccdType()));
+            step.putItem(BuiltinROIKey, (gmosParams.builtinROI()));
+            step.putItem(DisperserKey, (gmosParams.grating().displayValue()));
+            GmosCommonType.Binning xbin = GmosCommonType.Binning.getBinningByValue(gmosParams.spectralBinning());
+            GmosCommonType.Binning ybin = GmosCommonType.Binning.getBinningByValue(gmosParams.spatialBinning());
+            step.putItem(CcdXBinning, xbin);
+            step.putItem(CcdYBinning, ybin);
+            step.putItem(AmpGain, (gmosParams.ampGain()));
+            if (gmosParams.ccdType().displayValue().equals("E2V")) {
+                step.putItem(AmpCount, GmosCommonType.AmpCount.SIX);
+            } else {
+                step.putItem(AmpCount, GmosCommonType.AmpCount.TWELVE);
```

Comment:
I guess `AmpCount.THREE` is only kept around for legacy reasons.
So the assumption is if not `E2V` then `Hamamatsu`?

Reply:
Yes, in the ITC we only allow for the maximum number of amplifiers, so it’s either E2V with 6 amps, or Hamamatsu with 12 amps. I will change this to make it more specific, thanks.